src/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,445 @@
+  flex-direction: ${props =>
+    props.hasFile ? "row" : props.size === "sm" ? "row" : "column"};
+  align-items: center;
+  justify-content: ${props =>
+    props.hasFile ? "space-between" : props.size === "sm" ? "space-between" : "center"};
+  gap: 12px;
+  cursor: ${props => (props.hasFile ? "default" : "pointer")};
+  transition: all 0.2s ease;
+
+  ${props =>
+    !props.hasFile &&
+    css`
+      border-style: dashed;
+      border-color: ${({ theme }) => theme.click.card.primary.color.stroke.default};
+
+      &:hover,
+      &:focus {
+        background-color: ${({ theme }) =>
+          theme.click.card.secondary.color.background.hover};
+
+        button {
+          background-color: ${({ theme }) =>
+            theme.click.card.secondary.color.background.hover};
+        }
+      }
+
+      ${props.isDragging &&
+      css`
+        background-color: ${({ theme }) =>
+          theme.click.card.secondary.color.background.hover};
+        border-color: #9ca3af;
+      `}
+    `}
+
+  ${props =>
+    props.isError &&
+    css`
+      background-color: ${({ theme }) => theme.click.alert.color.background.danger};
+      border: none;
+    `}
+`;
+
+const DocumentIcon = styled(Icon)`
+  svg {
+    width: 32px;
+    height: 32px;
+    color: ${({ theme }) => theme.global.color.text.default};
+  }
+`;
+
+const UploadIcon = styled(Icon)`
+  svg {
+    width: 32px;
+    height: 32px;
+    ${({ theme }) =>
+      theme.name === "dark"
+        ? `
+    color: ${theme.global.color.text.default};
+  `
+        : `
+    color: ${theme.global.color.text.default};
+  `}
+  }
+`;
+
+const UploadText = styled.div<{ size: "sm" | "md"; hasFile: boolean }>`
+  text-align: ${props => (props.hasFile || props.size === "sm" ? "left" : "center")};
+  ${props =>
+    (props.hasFile || props.size === "sm") &&
+    css`
+      flex: 1;
+      margin-left: 12px;
+    `}
+
+  ${props =>
+    !props.hasFile &&
+    props.size === "md" &&
+    css`
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      width: 100%;
+    `}
+`;
+
+const FileInfoHeader = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 12px;
+  width: 100%;
+`;
+
+const FileInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+  flex: 1;
+`;
+
+const FileDetails = styled.div`
+  display: flex;
+  gap: 4px;
+  border: none;
+`;
+
+const FileActions = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: auto;
+  border: none;
+  background-color: transparent;
+  cursor: pointer;
+`;
+
+const ProgressContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 8px;
+  margin-top: 4px;
+  width: 100%;
+  padding-left: 4px;
+  padding-right: 8px;
+`;
+
+const ProgressBarContainer = styled.div`
+  width: 100%;
+  height: 4px;
+  background-color: #e5e7eb;
+  border-radius: 2px;
+  flex: 1;
+  overflow: hidden;
+`;
+
+const ProgressBar = styled.div<{ progress: number }>`
```

Comment:
I'm not sure, but I think props that are only used in styled components should have `$` as prefix, cc @vineethasok ?

Reply:
Correct, it will throw this warning locally:

Warning: React does not recognize the `progress` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `progress` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
div

This comment also applies to the prop names for UploadArea